app/controllers/api/vcs.php

```diff
@@ -458,9 +458,9 @@
         $github->initializeVariables($providerInstallationId, $privateKey, $githubAppId);
 
         $owner = $github->getOwnerName($providerInstallationId);
-        $repositoryName = $github->getRepositoryName($providerRepositoryId);
-
-        if (empty($repositoryName)) {
+        try {
+            $repositoryName = $github->getRepositoryName($providerRepositoryId);
```

Comment:
To make sure our code is not affected by edge cases, you should still check if the repository name is empty and then through the same error accordingly.

Reply:
I see same try/catch in multiple places. Lets make sure to address it everywhere.